### PR TITLE
Enable api extractor

### PR DESCRIPTION
**Pull request by joheredi**
Incorporating API extractor and creating CI yml for pipeline

### CODE REVIEW

File: sdk/apimanagement/apimanagement-custom-widget-scaffolder/review/apimanagement-custom-widget-scaffolder.api.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+## API Report File for "@azure-tools/apimanagement-custom-widget-scaffolder"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+// @public (undocumented)
+export function generateProject(widgetConfig: TWidgetConfig, deployConfig: TDeployConfig, { openUrl }?: TMiscConfig): Promise<void>;
+
+// @public (undocumented)
```
Comment (joheredi) [inline sdk/apimanagement/apimanagement-custom-widget-scaffolder/review/apimanagement-custom-widget-scaffolder.api.md:10]: Make sure to add TSDocs for all the places where (undocumented) shows up. The TS Doc needs to go in their respective files

File: sdk/apimanagement/apimanagement-custom-widget-scaffolder/review/apimanagement-custom-widget-scaffolder.api.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+export interface TCustomWidgetConfig extends TWidgetConfig {
+    // (undocumented)
+    deployed?: string;
+    // (undocumented)
+    override?: string | boolean;
```
Comment (joheredi) [inline sdk/apimanagement/apimanagement-custom-widget-scaffolder/review/apimanagement-custom-widget-scaffolder.api.md:15]: What string values do these take? Their name sound like booleans so I wonder if we can just define them as a boolean?

File: sdk/apimanagement/apimanagement-custom-widget-scaffolder/review/apimanagement-custom-widget-scaffolder.api.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+// @public (undocumented)
+export interface TDeployConfig {
```
Comment (joheredi) [inline sdk/apimanagement/apimanagement-custom-widget-scaffolder/review/apimanagement-custom-widget-scaffolder.api.md:19]: What does the T prefix mean? I wonder if we can remove it?

File: sdk/apimanagement/apimanagement-custom-widget-scaffolder/review/apimanagement-custom-widget-scaffolder.api.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    // (undocumented)
+    managementApiEndpoint: string;
+    // (undocumented)
+    resourceId: string;
+}
+
+// @public (undocumented)
+export interface TMiscConfig {
+    // (undocumented)
+    openUrl?: string;
```
Comment (joheredi) [inline sdk/apimanagement/apimanagement-custom-widget-scaffolder/review/apimanagement-custom-widget-scaffolder.api.md:31]: What is this URL used for?

File: sdk/apimanagement/apimanagement-custom-widget-scaffolder/src/index.ts [hunk 5]
```diff
@@ -1,2 +1,22 @@
-export {generateProject, displayNameToName} from "./scaffolding"
-export type {TWidgetRuntimeConfig as TCustomWidgetConfig} from "./scaffolding"
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * This is the ModelsRepositoryClient Library for JavaScript.
+ *
+ * @remarks
+ * TODO - add description
```
Comment (joheredi) [inline sdk/apimanagement/apimanagement-custom-widget-scaffolder/src/index.ts:8]: Make sure to fill in these 2 fields

File: sdk/apimanagement/apimanagement-custom-widget-scaffolder/src/scaffolding.ts [hunk 6]
```diff
@@ -39,16 +40,22 @@ export interface TMiscConfig {
 
 export type TConfigs = TWidgetConfig | TDeployConfig | TMiscConfig
 
+/**
+ * @internal
```
Comment (joheredi) [inline sdk/apimanagement/apimanagement-custom-widget-scaffolder/src/scaffolding.ts:44]: This is used when the function is exported from a module but it is not exposed publicly in the index.ts. @internal will hide it from the api-extractor output.

Please go through all the interfaces in the API Extractor md file and see if there are any interfaces that can be hiden similarly to this function

File: sdk/apimanagement/apimanagement-custom-widget-scaffolder/src/scaffolding.ts [hunk 7]
```diff
@@ -77,7 +84,6 @@ export async function generateProject(
     }
 
     const relativePath = file
-      .replace(/\//g, "\\")
```
Comment (joheredi) [inline sdk/apimanagement/apimanagement-custom-widget-scaffolder/src/scaffolding.ts:80]: This seems to have been causing the weird path problem

Comment (joheredi) [reply sdk/apimanagement/apimanagement-custom-widget-scaffolder/src/scaffolding.ts:80]: Added a check for handling windows paths

**Comment by joheredi** [comment]
@JMach1  after merging this PR you will need to run

```bash
rushx format
rushx lint --fix
```

And fix the remaining eslint issues that will show up in the terminal.

File: sdk/apimanagement/apimanagement-custom-widget-scaffolder/review/apimanagement-custom-widget-scaffolder.api.md (hunk 3 above)
Comment (JMach1) [reply sdk/apimanagement/apimanagement-custom-widget-scaffolder/review/apimanagement-custom-widget-scaffolder.api.md:19]: I make all types start with a T so it's immediately clear it's a type/interface

File: sdk/apimanagement/apimanagement-custom-widget-scaffolder/review/apimanagement-custom-widget-scaffolder.api.md (hunk 4 above)
Comment (JMach1) [reply sdk/apimanagement/apimanagement-custom-widget-scaffolder/review/apimanagement-custom-widget-scaffolder.api.md:31]: It's for specifying URL the scaffolded project opens on dev env startup.

File: sdk/apimanagement/apimanagement-custom-widget-scaffolder/review/apimanagement-custom-widget-scaffolder.api.md (hunk 2 above)
Comment (JMach1) [reply sdk/apimanagement/apimanagement-custom-widget-scaffolder/review/apimanagement-custom-widget-scaffolder.api.md:15]: deployed is timestamp when it was deployed and override optionally contains from where is the content overridden from

**Pull request merged**